src/StaticMock/Mock.cs

```diff
@@ -27,7 +27,7 @@ public static IFuncMockService Setup(Type type, string methodName, Action action
                 throw new ArgumentNullException(nameof(action));
             }
 
-            var methodToReplace = type.GetMethod(methodName);
+            var methodToReplace = type.GetMethod(methodName, BindingFlags.Instance | BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic);
```

Comment:
I think we need to add overload for setup method where we can put BindingFlags as parameter like Microsoft implementation of GetMethod(String, BindingFlags), GetMethod(String) (https://docs.microsoft.com/en-us/dotnet/api/system.type.getmethod?view=net-5.0#overloads). Because, I think, it isn't obviously that our method Setup(Type type, string methodName, Action action) can setup private methods or properties. User must put BindingFlags by himself, if he want to mock private methods or properties

Reply:
Yes, I agree with you. I'll make time to revise it tomorrow morning.Thankyou